src/plugins/intel_gpu/src/graph/program.cpp

```diff
@@ -177,14 +177,18 @@ void program::init_program() {
 }
 
 void program::init_primitives() {
+    // Register implementations in order of their selection priority: common, OCL, oneDNN, CPU
```

Comment:
@vladimir-paramuzov, yeah, moving `onednn` impls initialization before `ocl` will be more aligned with priorities, but it will affect some amount of simple tests where optimize_data property is `false` and impl type remains `any` after all optimization passes, and then oneDNN impls will be used for such operations (including iGPU). It will reduce cldnn's operations test coverage and cause redundant tests for oneDNN even in cases we don't use it at all. That is why I left `ocl`/`onednn` priority unchanged